packages/commonwealth/server/routes/updateChainCustomDomain.ts

```diff
@@ -0,0 +1,39 @@
+import { Response, NextFunction } from 'express';
+import { TypedRequestBody } from '../types';
+import { DB } from '../database';
+
+type updateCustomDomainReq = {
+  secret: string;
+  chain_id: string;
+  custom_domain: string;
+};
+
+const updateChainCustomDomain = async (
+  models: DB,
+  req: TypedRequestBody<updateCustomDomainReq>,
+  res: Response,
+  next: NextFunction
+) => {
+  if (!req.body) {
+    return next(new Error('invalid request body'));
+  }
+  // Verify Chain Exists
+  const { chain_id, custom_domain, secret } = req.body;
+  const chain = await models.Chain.findOne({
+    where: { id: chain_id },
+  });
+  if (!chain) return next(new Error('Chain not found.'));
+
+  // Check secret
+  if (process.env.CUSTOM_DOMAIN_UPDATE_SECRET !== secret) {
+    return next(new Error('Invalid secret.'));
+  }
+
+  // Update Custom Domain
+  chain.custom_domain = custom_domain;
```

Comment:
We should do some validation on custom_domain, to ensure it's formatted in a valid way: xxx.yyy[.zzz], no http/https prefix, no invalid characters for a URL. Google around to find a good regex for a valid hostname.

Reply:
we have email regex checks in the app, ctrl+f `validEmailRegex`

Reply:
actually @gdjohnson wrote a validURL regex helper `validURL(str)` which we should use here